ls1intum/Artemis
# `Discussion`: Improve emoji-bar and answer option design

### Checklist
#### General


#### Client

### Motivation and Context

1. In the current version of Posts and AnswerPosts through Artemis, the "emoji-bar" is located at the bottom right corner of the messages. Although users are able to show their reactions to others without any issues, readers have to focus on the end of the message to realize the emotional reaction of the author. This is inconvenient and contradicts the standards of today's written communication platforms.
2. Another issue with Posts especially on the Discussions page is that the height of Posts together with their AnswerPosts is much longer than they are needed. This causes readers to perform lots of redundant scrolling to reach information.




### Description

1. In order to let users easily be aware of the communicated emotions of the authors, we relocate the emoji bar to the left-hand side of Post and AnswerPost footers. Since Post tags are located in this place, we relocate Post tags to the initial place of the emoji bar, which is the right-hand side of Post footer. Please note that AnswerPosts do not have tags.
2. To reduce the minimum Post + AnswerPost height, we rework the toggle buttons "Start Discussion", "Show {{x}} Replies" and "Hide Replies" buttons, by integrating them within the emoji-bar, with a toggleable button that has a discussion-bubble and closing icon.



### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Lecture with Discussions enabled

1. Log in to Artemis
3. Navigate to Course
4. Select Discussions page
5. Create a new Post with tags.
7. Make sure that the emoji bar is displayed correctly on the left-hand side of the footer
8. Make sure that the tags bar is displayed on the right-hand side of the footer
9. Click the button "Start Discussion" found within the emoji bar and post a new Answer
10. Make sure that the "Start Discussion" button is replaced with "Hide Replies" button which is toggled to indicate that AnswerPosts are displayed
11. Click on the "Hide Replies" button
12. Make sure that now "Show {{x}} Replies" button is available within the emoji bar
13. Go to step 1 and perform each step for an Exercise specific discussion, by navigating to a random exercise

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| answer-post.component.ts | 100% | 100% |
| post-footer.component.ts | 88.23% | 100% |
| post-reactions-bar.component.ts | 83.33% | 95.45% |
| posting-thread.component.ts | 100% | 100% |



### Screenshots


![image](https://user-images.githubusercontent.com/67344158/168480960-6329daea-8824-4c45-94c3-e3fc7fc5e98a.png)
![image](https://user-images.githubusercontent.com/67344158/168480970-e5096eb3-d80b-4da7-8127-b7c661d72785.png)
![image](https://user-images.githubusercontent.com/67344158/168480986-906874d5-ef0c-4d96-a0e3-079816d6e7ed.png)
